uswds/uswds
# USWDS - POAM: November ‘24

# Summary

Updated vulnerable dependencies, updated indirect dependency for elliptic
## Breaking change

This is not a breaking change.

## Related issue

uswds/uswds-team#413

## Preview link

[Preview link:](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cb-poam-nov-24/?path=/story/design-tokens-fonts--fonts)


## Major changes

Before:

```node
57 vulnerabilities (28 moderate, 29 high)
```

After:

```node
57 vulnerabilities (28 moderate, 29 high)
```

Node.js updated from `v 20.18.1 to 22.11.0` to get the latest LTS version
`(v22.11.0   (Latest LTS: Jod))
`
## Testing and review

1. Pull down and run `npm install`
2. Run `node -v` and check major version is 22 
3. Confirm no installation errors
4. Run Compile scripts
5. Confirm no build errors
6. Run `npm run test`
7. Confirm no testing or linting errors 
9. Start local or visit preview link
10. Confirm there are no visual regressions

## Dependency updates

| **Dependency name** | **Old version** | **New version** |
| --- | --- | --- |
| @babel-core | 7.25.7 | 7.26.0 |
| @babel/preset-env  | 7.25.7 | 7.26.0 |
| axe-core | 4.10.0 | 4.10.2
| cross-spawn | 7.0.3 | 7.0.6
| html-webpack-plugin | 5.6.0 | 5.6.3
| mocha | 10.7.3 | 10.8.2
| postcss | 8.4.47 | 8.4.49
| sass | 1.79.4 | 1.81.0
| sass-embedded | 1.79.4 | 1.81.0
| snyk | 1.1293.1 | 1.1294.1
| stylelint | 16.9.0 | 16.10.0
| typescript | 5.6.2 | 5.7.2
| webpack | 5.95.0 | 5.96.1
